exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/RestClientFactory.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.exporter.opensearch;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.impl.nio.reactor.IOReactorConfig;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestClientBuilder;
+
+final class RestClientFactory {
+  private static final RestClientFactory INSTANCE = new RestClientFactory();
+
+  private RestClientFactory() {}
+
+  /**
+   * Returns a {@link RestClient} instance based on the given configuration. The URL is parsed as a
+   * comma separated list of "host:port" formatted strings. Authentication is supported only as
+   * basic auth; if there is no authentication present, then nothing is configured for it.
+   */
+  static RestClient of(final OpensearchExporterConfiguration config) {
+    return of(config, false);
+  }
+
+  /**
+   * Returns a {@link RestClient} instance based on the given configuration. The URL is parsed as a
+   * comma separated list of "host:port" formatted strings. Authentication is supported only as
+   * basic auth; if there is no authentication present, then nothing is configured for it.
+   *
+   * @param config the exporter configuration
+   * @param allowAllSelfSignedCertificates if set to true ALL self-signed certificates will be
```

Comment:
:upside_down_face: Just nitpicking, we can add a comma here, but my English grammar knowledge might be wrong.

```suggestion
   * @param allowAllSelfSignedCertificates if set to true, ALL self-signed certificates will be
```